shared/src/main/scala/kinesis4cats/producer/metrics/MetricsReporter.scala

```diff
@@ -0,0 +1,153 @@
+package kinesis4cats.producer.metrics
+
+import scala.concurrent.duration._
+
+import cats.Applicative
+import cats.effect.Async
+import cats.effect.Fiber
+import cats.effect.Resource
+import cats.effect.syntax.all._
+import cats.syntax.all._
+import fs2.Chunk
+import fs2.concurrent.Channel
+import org.typelevel.log4cats.StructuredLogger
+
+import kinesis4cats.compat.retry._
+import kinesis4cats.logging._
+
+abstract class MetricsReporter[F[_], PutRes](
```

Comment:
Setting this up very similarly to the `FS2Producer`, so that we can have a defined, configurable rate of metrics publishing.